src/kernel/filesystem/fat32.zig

```diff
@@ -0,0 +1,1756 @@
+                    .fs = fs,
+                    .mount = null,
+                },
+            };
+
+            fs.* = .{
+                .open = open,
+                .close = close,
+                .read = read,
+                .write = write,
+                .instance = &fat32_fs.instance,
+                .getRootNode = getRootNode,
+            };
+
+            // Record the relevant information
+            const fat_config = FATConfig{
+                .bytes_per_sector = boot_sector.bytes_per_sector,
+                .sectors_per_cluster = boot_sector.sectors_per_cluster,
+                .reserved_sectors = boot_sector.reserved_sectors,
+                .hidden_sectors = boot_sector.hidden_sectors,
+                .total_sectors = boot_sector.total_sectors,
+                .sectors_per_fat = boot_sector.sectors_per_fat,
+                .root_directory_cluster = boot_sector.root_directory_cluster,
+                .fsinfo_sector = boot_sector.fsinfo_sector,
+                .backup_boot_sector = boot_sector.backup_boot_sector,
+                .has_fs_info = has_fs_info,
+                .number_free_cluster = number_free_cluster,
+                .next_free_cluster = next_free_cluster,
+                .cluster_end_marker = cluster_end_marker,
+            };
+
+            fat32_fs.* = .{
+                .fs = fs,
+                .allocator = allocator,
+                .instance = 32,
+                .root_node = .{
+                    .node = root_node,
+                    .cluster = fat_config.root_directory_cluster,
+                },
+                .fat_config = fat_config,
+                .stream = stream,
+                .opened_files = AutoHashMap(*const vfs.Node, *OpenedInfo).init(allocator),
+            };
+
+            return fat32_fs;
+        }
+    };
+}
+///
+/// Initialise a FAT32 filesystem. This will take a stream that conforms to the reader(), writer(),
+/// and seekableStream() interfaces. For example, FixedBufferStream or File. A pointer to this is
+//// allowed. This will allow the use of different devices such as Hard drives or RAM disks to have
+/// a FAT32 filesystem abstraction. We assume the FAT32 will be the only filesystem present and no
+/// partition schemes are present. So will expect a valid boot sector.
+///
+/// Arguments:
+///     IN allocator: *Allocator - An allocator.
+///     IN stream: anytype       - A stream this is used to read and seek a raw disk or memory to
+///                                be parsed as a FAT32 filesystem. E.g. a FixedBufferStream.
+///
+/// Return: *Fat32FS(@TypeOf(stream))
+///     A pointer to a FAT32 filesystem.
+///
+/// Error: Allocator.Error || Fat32FS(@TypeOf(stream)).Error
+///     Allocator.Error - If there isn't enough memory to create the filesystem.
+///
+pub fn initialiseFAT32(allocator: *Allocator, stream: anytype) (Allocator.Error || ErrorSet(@TypeOf(stream)) || Fat32FS(@TypeOf(stream)).Error)!*Fat32FS(@TypeOf(stream)) {
```

Comment:
Do we need this when we have Fat32FS.init?

Reply:
It just makes the API easier to use. Plus this is how I have seen it in the std lib.